text/0000-anon-lifetime.md

```diff
@@ -0,0 +1,69 @@
+- Feature Name: anon_lifetime
+- Start Date: 2015-06-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Allow using an undeclared '_ wherever an explicit lifetime can be used, but is
+optional, such as function argument/return types and any path inside a function.
+
+# Motivation
+
+During the addition of 'tcx in the compiler (denoting the lifetime of the
+arenas used to allocate all the type-system information in the compiler,
+via the type context aka `ty::ctxt` aka `tcx`), I had attempted to replace
+`&'a ty::ctxt<'tcx>` with a clean `TyCx<'a, 'tcx>` wrapper.
+When both `'a` and `'tcx` are either present or missing, it works very well,
+but the common case is `&ty::ctxt<'tcx>`, which cannot be represented by
+`TyCx<'a, 'tcx>` without adding a new explicit lifetime parameter (`'a`)
+to the appropriate function or impl block.
+
+I've recently done this as part of rust-lang/rust#26575, but it wasn't easy
+or pretty and any attempts at adding another lifetime to `TyCx` (necessary
+for splitting type contexts for performance and possibly parallelism in the
+future) would result in more noise at almost every use site of `TyCx`.
+
+# Detailed design
+
+In `resolve_lifetime`: if the lifetime to be resolved matches "'_", store
```

Comment:
`"'_"` should be in backticks so GitHub doesn’t render the text after it italicised.